engines/director/cast.cpp

```diff
@@ -711,9 +789,316 @@ void Cast::loadCast() {
+		} else {
+			// The size stored in the memory map (_resources array), as well as the resource itself is the size
+			// excluding the 'CASt' header and the entry of size itself. Adding 8 to compensate for that
+			castSize = _castArchive->getResourceSize(MKTAG('C', 'A', 'S', 't'), it) + 8;
+		}
+
+		debugC(5, kDebugSaving, "Cast::saveCast()::Saving 'CASt' resource, id: %d, size: %d", id, castSize);
+		byte *dumpData = (byte *)calloc(castSize, sizeof(byte));
+
+		Common::SeekableMemoryWriteStream *writeStream = new Common::SeekableMemoryWriteStream(dumpData, castSize);
+        CastType type = kCastTypeAny;
+
+		if (!_loadedCast->contains(id)) {
+            writeStream->writeUint32LE(MKTAG('C', 'A', 'S', 't'));
+            Common::SeekableReadStreamEndian *stream = getResource(MKTAG('C', 'A', 'S', 't'), it);
+            uint32 size = stream->size();           // This is the size of the Resource without header and size entry itself
+            writeStream->writeUint32LE(size);
+            writeStream->writeStream(stream);
+
+            delete stream;
+		} else {
+			CastMember *target = _loadedCast->getVal(id);
+            type = target->_type;
+
+			switch (target->_type) {
+            case kCastPalette:
+            case kCastBitmap:
+            case kCastText:
+            case kCastButton:
+            case kCastDigitalVideo:
+            case kCastFilmLoop:
+            case kCastMovie:
+            case kCastSound:
+            case kCastRichText:
+            case kCastLingoScript:
+            case kCastShape:
+				target->writeCAStResource(writeStream, 0);
+				break;
+			default:
+				target->writeCAStResource(writeStream, 0, it);
+				break;
+			}
+		}
+
+        dumpFile(castType2str(type), res.index, MKTAG('C', 'A', 'S', 't'), dumpData, castSize);
+		delete writeStream;
+	}
 
 }
 
+void Cast::writeCastInfo(Common::MemoryWriteStream *writeStream, uint32 castId) {
+	// The structure of the CastMemberInfo is as follows:
+	// First some headers: offset, unkonwns and flags, and then a count of strings to be read
+	// (These strings contain properties of the cast member like filename, script attached to it, name, etc.)
+	// After the header, we have the next data is the lengths of the strings,
+	// The first int is 0, the second int is 0 + length of the first string, the third int is 0 + length of first string + length of second string and so on
+	// After the lengths of the strings are the actual strings
+
+	if (!_castsInfo.contains(castId)) {
+		return;
+	}
+	CastMemberInfo *ci = _castsInfo[castId];
+
+	InfoEntries castInfo;
+
+	castInfo.unk1 = ci->unk1;
+	castInfo.unk2 = ci->unk2;
+	castInfo.flags = ci->flags;
+	castInfo.scriptId = ci->scriptId;
+	castInfo.strings.resize(ci->count);
+
+	for (int i = 1; i <= ci->count; i++) {
+		castInfo.strings[i - 1].len = getCastInfoStringLength(i, ci);
+	}
+
+	for (int i = 1; i <= ci->count; i++) {
+		if (!castInfo.strings[i - 1].len) {
+			continue;
+		}
+
+		switch (i) {
```

Comment:
```suggestion
		switch (i + 1) {
```
Though, I would just shift the `case` labels